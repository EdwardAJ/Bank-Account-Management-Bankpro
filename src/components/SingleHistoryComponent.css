.card {
    height: 140px;
    width: 225px;
    border-radius: 10px;
    border-style: solid;
    border-color: white;
    border-width: 1.35px;
    margin-bottom: 20px;
    margin-right: 15px;
}

.card-content {
    color: white;
    margin-left: 20px;
}

.type {
    font-weight: 600;
    font-size: 1rem;
}

.amount {
    font-weight: 700;
    font-size: 2rem;
    margin-top: -20px;
}

.account-number {
    display: flex;
    /* font-weight: 600; */
    margin-top: -32.5px;
}

.time-date {
    text-align: right;
    font-size: 0.7rem;
    margin-right: 20px;
    margin-top: -2.5px;
}