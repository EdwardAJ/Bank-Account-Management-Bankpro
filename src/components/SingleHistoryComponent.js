import React from 'react';
import './SingleHistoryComponent.css';

export default class SingleHistoryComponent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="card">
                <div className="card-content">
                    <p className="type"> { this.props.type } </p>
                    <p className="amount"> { this.props.amount } </p>
                    <p className="account-number"> { this.props.accountNumber } </p>
                    <p className="time-date"> { this.props.timeDate } </p>
                </div>
            </div>
        );
    }
}