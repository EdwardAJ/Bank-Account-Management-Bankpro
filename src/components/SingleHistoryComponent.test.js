import React from 'react';
import { shallow } from 'enzyme';
import SingleHistoryComponent from './SingleHistoryComponent';

describe("SingleHistoryComponent", () => {
    it("should render my component", () => {
        const props = {
            type: "Debit",
            amount: "40000",
            accountNumber: "13517115",
            timeDate: "2019.11.12 07:08:09"
        };
        const component = shallow(<SingleHistoryComponent {...props} />);
        expect(component.getElements()).toMatchSnapshot();
    });
});