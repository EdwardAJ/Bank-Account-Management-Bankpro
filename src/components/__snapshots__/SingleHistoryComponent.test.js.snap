exports[`SingleHistoryComponent should render my component 1`] = `
Array [
  <div
    className="card">
    <div
        className="card-content">
        <p
            className="type">
             
            Debit
             
        </p>
        <p
            className="amount">
             Rp
            40000
             
        </p>
        <p
            className="account-number">
             No Rek: 
            13517115
             
        </p>
        <p
            className="time-date">
             
            2019.11.12 07:08:09
             
        </p>
    </div>
</div>,
]
`;
